docs/src/testing.md

```diff
@@ -0,0 +1,81 @@
+We could add a test to ECOS.jl, but that would only stop us from re-introducing the bug to ECOS.jl in the future.
+Instead if we add a test here, then all solvers (e.g., SCS.jl, Gurobi.jl, Mosek.jl, ...) will also check that they handle a double optimize call!
+
+For this test, we care about correctness, rather than performance.
+We don't expect solvers to efficiently decide that they have already solved the problem,
+only that calling `optimize!` twice doesn't throw an error or give the wrong answer.
+
+To resolve this issue, there are two steps:
+
+1. Since the double-optimize error involves solving an optimization problem,
+add a new test to [src/Test/UnitTests/solve.jl](https://github.com/jump-dev/MathOptInterface.jl/blob/master/src/Test/UnitTests/solve.jl).
+The test should be something like
+
+```julia
+function solve_twice(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    x = MOI.add_variable(model)
+    c = MOI.add_constraint(model, MOI.SingleVariable(x), MOI.GreaterThan(1.0))
+    MOI.set(model, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+    MOI.set(model, MOI.ObjectiveFunction{MOI.SingleVariable}(), MOI.SingleVariable(x))
+    if config.solve
+        MOI.optimize!(model)
+        MOI.optimize!(model)
+        MOI.get(model, MOI.TerminationStatus()) == MOI.OPTIMAL
+        MOI.get(model, MOI.VariablePrimal(), x) == 1.0
+    end
+end
+unittests["solve_twice"] = solve_twice
+```
+
+2. Add a test for the test you just wrote (We test the tests!)
+
+    a. Add the name of the test ("solve_twice") to the end of the array in `MOIT.unittest(...)` ([link](https://github.com/jump-dev/MathOptInterface.jl/blob/7543afe4b5151cf36bbd18181c1bb5c83266ae2f/test/Test/unit.jl#L51-L52)).
+
+    b. Add a test for the test towards the end of the "Unit Tests" test set ([link](https://github.com/jump-dev/MathOptInterface.jl/blob/7543afe4b5151cf36bbd18181c1bb5c83266ae2f/test/Test/unit.jl#L394)). The test should look something like
+
+```julia
+@testset "solve_twice" begin
+    MOIU.set_mock_optimize!(mock,
+        (mock::MOIU.MockOptimizer) -> MOIU.mock_optimize!(
+            mock,
+            MOI.OPTIMAL,
+            (MOI.FEASIBLE_POINT, [1.0]),
+        ),
+        (mock::MOIU.MockOptimizer) -> MOIU.mock_optimize!(
+            mock,
+            MOI.OPTIMAL,
+            (MOI.FEASIBLE_POINT, [1.0]),
+        )
+    )
+    MOIT.solve_twice(mock, config)
+end
+```
+
+In the above `mock` is a `MOI.Utilities.MockOptimizer` that is defined earlier in the file.
+In this test, `MOIU.set_mock_optimize!` loads `mock` with two results. Each says
+that the [`MOI.TerminationStatus`](@ref) is `MOI.OPTIMAL`, that the
+[`MOI.PrimalStatus`](@ref) is `MOI.FEASIBLE_POINT`, and that there is one
+variable with a `MOI.VariableValue` or `1.0`.
+
+To run the tests:
+
+```
+julia # with 1.5
+julia> ]
+(@v1.5) pkg> dev ECOS
+(@v1.5) pkg> dev MathOptInterface
+
+Make changes in ~/.julia/dev/ECOS or ~/.julia/dev/MathOptInterface
```

Comment:
This bit isn't really `To run the tests:`. 

Perhaps step 1 should be `dev`ing the package, and the final step can be running the tests.

Reply:
done